cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -483,6 +484,29 @@ std::string FormatRange(int64_t start, int64_t length) {
   return ss.str();
 }
 
+// An AWS RetryStrategy that wraps a provided arrow::fs::S3RetryStrategy
+class WrappedRetryStrategy : public Aws::Client::RetryStrategy {
+ public:
+  explicit WrappedRetryStrategy(const std::shared_ptr<S3RetryStrategy>& s3_retry_strategy)
+      : s3_retry_strategy_(s3_retry_strategy) {}
+
+  bool ShouldRetry(const Aws::Client::AWSError<Aws::Client::CoreErrors>& error,
+                   long attempted_retries) const override {
```

Comment:
Hmm, sorry for the back-and-forth, it appears the NOLINT was because we can't use `long` here. Given that this is a third party interface we're implementing, I think it should be OK to place a `// NOLINT runtime/int` to override solely this warning (rather than all of them).

Below, though, we should return int64_t from the Arrow S3RetryStrategy::CalculateDelayBeforeNextRetry instead of long.

Reply:
(I'm looking at the lint failures here: https://github.com/apache/arrow/pull/10877/checks?check_run_id=3255790318#step:7:4223)